website/static/js/addonSettings.js

```diff
@@ -25,20 +26,28 @@ var ExternalAccount = function(data) {
     });
 
     self.deauthorizeNode = function(node) {
-        var url = node.urls.deauthorize;
-        var request = $.ajax({
-            url: url,
-            type: 'DELETE'
-        });
-        request.done(function(data) {
-            self.connectedNodes.remove(node);
-        });
-        request.fail(function(xhr, status, error) {
-            Raven.captureMessage('Error deauthorizing node: ' + node.id, {
-                url: url, status: status, error: error
-            });
+        bootbox.confirm({
+            title: 'Remove addon?',
+            message: 'Are you sure you want to remove the ' + self.provider_name + ' authorization from this project?',
+            callback: function (confirm) {
+                if (confirm) {
+                    var url = node.urls.deauthorize;
```

Comment:
In order to make this more testable, could you separate this request code into another method that can be tested independently of the confirmation logic?